### ISSUE TITLE
pyi mode

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright-playground#11]

**Comment by jorenham** [comment]
*this is a repost from #29*

---

it can be quite a bit easier to experiment with typing stuff in .pyi stubs than in .py modules, because

__future__.annotations or stringified annotations aren't needed at all (and illegal according to flake8-pyi in ruff)
typing_extensions and _typeshed are always available
in the same way, you could "install libraries" by putting their stubs in the custom typeshed/stdlib
reveal_locals can be used
in the same way, you could easily add you own builtins (e.g. from basedtyping or optype)
it's could be more obvious for newcomers that there isn't a runtime
locally I've been playing around in .pyi files this way for quite a bit, and I can't think of any downsides

so I think it could be helpful to either completly switch to .pyi, or leave that choice up to the user (url-based might be an easy solution)

**Comment by jorenham** [comment]
I'm guessing that this is the relevant code then
https://github.com/DetachHead/basedpyright-playground/blob/main/client/LspClient.ts#L57

**Comment by DetachHead** [comment]
yeah i think so, shouldn't be too hard to implement assuming all we need to do is andd an option that changes that filename
